smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpProtocolClientGenerator.kt

```diff
@@ -208,16 +213,20 @@ abstract class HttpProtocolClientGenerator(
         val hasOutputStream = outputShape.map { it.hasStreamingMember(ctx.model) }.orElse(false)
         val inputVariableName = if (inputShape.isPresent) "input" else KotlinTypes.Unit.fullName
 
-        if (hasOutputStream) {
-            writer
-                .addImport(RuntimeTypes.Http.Operation.execute)
-                .write("return op.#T(client, #L, block)", RuntimeTypes.Http.Operation.execute, inputVariableName)
-        } else {
-            writer.addImport(RuntimeTypes.Http.Operation.roundTrip)
-            if (outputShape.isPresent) {
-                writer.write("return op.#T(client, #L)", RuntimeTypes.Http.Operation.roundTrip, inputVariableName)
+        writer.withBlock(
+            "return op.context.#T(op.context.#T) {",
+            "}",
+            RuntimeTypes.Tracing.Core.childTraceSpan,
+            RuntimeTypes.Http.Operation.sdkRequestId,
```

Comment:
still think it might be worthwhile to include the operation name

Reply:
👍 Yep I agree, just forgot.